In [1]:
import csv
f=open("guns.csv")
data=list(csv.reader(f))
data[:5]

[['',
  'year',
  'month',
  'intent',
  'police',
  'sex',
  'age',
  'race',
  'hispanic',
  'place',
  'education'],
 ['1',
  '2012',
  '01',
  'Suicide',
  '0',
  'M',
  '34',
  'Asian/Pacific Islander',
  '100',
  'Home',
  '4'],
 ['2', '2012', '01', 'Suicide', '0', 'F', '21', 'White', '100', 'Street', '3'],
 ['3',
  '2012',
  '01',
  'Suicide',
  '0',
  'M',
  '60',
  'White',
  '100',
  'Other specified',
  '4'],
 ['4', '2012', '02', 'Suicide', '0', 'M', '64', 'White', '100', 'Home', '4']]

In [2]:
headers=data[0]
data=data[1:]
headers


['',
 'year',
 'month',
 'intent',
 'police',
 'sex',
 'age',
 'race',
 'hispanic',
 'place',
 'education']

In [3]:
years = [ row[1] for row in data]
year_counts = {}
for year in years:
    if year in year_counts:
        year_counts[year]+=1
    else:
        year_counts[year]=1
        
print(year_counts)

{'2014': 33599, '2012': 33563, '2013': 33636}


In [10]:
import datetime
dates = []
for row in data:
    a = datetime.datetime(year=int(row[1]), month=int(row[2]), day = 1)
    dates.append(a)
    
print(dates[:5])
date_counts = {}
for date in dates:
    if date in date_counts:
        date_counts[date]+=1
    else:
        date_counts[date]=1
        
date_counts

[datetime.datetime(2012, 1, 1, 0, 0), datetime.datetime(2012, 1, 1, 0, 0), datetime.datetime(2012, 1, 1, 0, 0), datetime.datetime(2012, 2, 1, 0, 0), datetime.datetime(2012, 2, 1, 0, 0)]


{datetime.datetime(2012, 1, 1, 0, 0): 2758,
 datetime.datetime(2012, 2, 1, 0, 0): 2357,
 datetime.datetime(2012, 3, 1, 0, 0): 2743,
 datetime.datetime(2012, 4, 1, 0, 0): 2795,
 datetime.datetime(2012, 5, 1, 0, 0): 2999,
 datetime.datetime(2012, 6, 1, 0, 0): 2826,
 datetime.datetime(2012, 7, 1, 0, 0): 3026,
 datetime.datetime(2012, 8, 1, 0, 0): 2954,
 datetime.datetime(2012, 9, 1, 0, 0): 2852,
 datetime.datetime(2012, 10, 1, 0, 0): 2733,
 datetime.datetime(2012, 11, 1, 0, 0): 2729,
 datetime.datetime(2012, 12, 1, 0, 0): 2791,
 datetime.datetime(2013, 1, 1, 0, 0): 2864,
 datetime.datetime(2013, 2, 1, 0, 0): 2375,
 datetime.datetime(2013, 3, 1, 0, 0): 2862,
 datetime.datetime(2013, 4, 1, 0, 0): 2798,
 datetime.datetime(2013, 5, 1, 0, 0): 2806,
 datetime.datetime(2013, 6, 1, 0, 0): 2920,
 datetime.datetime(2013, 7, 1, 0, 0): 3079,
 datetime.datetime(2013, 8, 1, 0, 0): 2859,
 datetime.datetime(2013, 9, 1, 0, 0): 2742,
 datetime.datetime(2013, 10, 1, 0, 0): 2808,
 datetime.datetime(2013, 11,

In [18]:
sex_counts={}
race_counts={}
for row in data:
    sex,race = row[5],row[7]
    if sex in sex_counts:
        sex_counts[sex]+=1
    else:
        sex_counts[sex]=1
    if race in race_counts:
        race_counts[race]+=1
    else:
        race_counts[race]=1
print(sex_counts)
race_counts

{'M': 86349, 'F': 14449}


{'Asian/Pacific Islander': 1326,
 'Black': 23296,
 'Hispanic': 9022,
 'Native American/Native Alaskan': 917,
 'White': 66237}

# What I have learnt so far

The number of gun deaths in USA has not changed over the **Years [2012-2014]**. It is also doesn't change in *Months**

The victim of gun deaths are majorly **White people** followed by **Black people**

I think that the **Age of the victim** and the **Intent of the death** requires more scrutinizing.


In [19]:
f=open("census.csv","r")
census=list(csv.reader(f))
print(census)

[['Id', 'Year', 'Id', 'Sex', 'Id', 'Hispanic Origin', 'Id', 'Id2', 'Geography', 'Total', 'Race Alone - White', 'Race Alone - Hispanic', 'Race Alone - Black or African American', 'Race Alone - American Indian and Alaska Native', 'Race Alone - Asian', 'Race Alone - Native Hawaiian and Other Pacific Islander', 'Two or More Races'], ['cen42010', 'April 1, 2010 Census', 'totsex', 'Both Sexes', 'tothisp', 'Total', '0100000US', '', 'United States', '308745538', '197318956', '44618105', '40250635', '3739506', '15159516', '674625', '6984195']]


In [22]:
mapping = { 'Asian/Pacific Islander': 15159516 + 674625 ,
 'Black': 40250635,
 'Hispanic': 44618105,
 'Native American/Native Alaskan': 3739506 ,
 'White': 197318956 }

race_per_hundredk = {}
for k,v in race_counts.items():
    a = v / mapping[k]
    a = a * 100000
    race_per_hundredk[k] = a
    
print(race_per_hundredk)

{'Black': 57.8773477735196, 'Asian/Pacific Islander': 8.374309664161762, 'Native American/Native Alaskan': 24.521955573811088, 'Hispanic': 20.220491210910907, 'White': 33.56849303419181}


In [24]:
homicide_race_counts = {}
for row in data:
    if row[3] == "Homicide":
        if row[7] in homicide_race_counts:
            homicide_race_counts[row[7]]+=1
        else:
            homicide_race_counts[row[7]]=1
            
temp = {} #Because i am too lazy to type homicide_race_counts_per_hundredk everytime

for k,v in homicide_race_counts.items():
    a = v / mapping[k]
    a = a * 100000
    temp[k]=a
    
homicide_race_counts_per_hundredk = temp #Immediately regret my decision

homicide_race_counts_per_hundredk

{'Asian/Pacific Islander': 3.530346230970155,
 'Black': 48.471284987180944,
 'Hispanic': 12.627161104219914,
 'Native American/Native Alaskan': 8.717729026240365,
 'White': 4.6356417981453335}

# Takeaways #2 

### For every 100,000 people of a race 

**57.88** Black people were victims of gun deaths. Out of these, **48.47** were because of **Homicidial** reasons.

**20.22** Hispanic people were victims of gun deaths. Out of these, **12.63** were because of **Homicidial** reasons.

**33.57** White people were victims of gun deaths. Out of these, only **4.64** were because of **Homicidial** reasons.

